lang/c++/impl/Compiler.cc

```diff
@@ -314,16 +335,20 @@ static Field makeField(const Entity& e, SymbolTable& st, const string& ns)
     Object::const_iterator it = findField(e, m, "type");
     map<string, Entity>::const_iterator it2 = m.find("default");
     NodePtr node = makeNode(it->second, st, ns);
+    if (containsField(m, "doc")) {
+        node->setDoc(getDocField(e, m));
+    }
     GenericDatum d = (it2 == m.end()) ? GenericDatum() :
         makeGenericDatum(node, it2->second, st);
     return Field(n, node, d);
 }
 
-static NodePtr makeRecordNode(const Entity& e,
-    const Name& name, const Object& m, SymbolTable& st, const string& ns)
-{
+// Extended makeRecordNode (with doc).
+static NodePtr makeRecordNode(const Entity& e, const Name& name,
+                              const std::string* doc, const Object& m,
+                              SymbolTable& st, const std::string& ns) {
     const Array& v = getArrayField(e, m, "fields");
-    concepts::MultiAttribute<string> fieldNames;
+    concepts::MultiAttribute<std::string> fieldNames;
```

Comment:
Inside `cc` files we try to avoid using `std::` prefix and instead use plain `string` etc along with `using` directive. We never use `using` in header file. We follow the advice [59. Don’t write namespace usings in a header file or before an #include](https://www.safaribooksonline.com/library/view/c-coding-standards/0321113586/ch60.html)